src/main/java/duke/command/DoneCommand.java

```diff
@@ -0,0 +1,37 @@
+package duke.command;
+
+import duke.task.Task;
+import duke.task.TaskList;
+
+public class DoneCommand extends Command {
+    private int idx;
+
+    /**
+     * Constructor
+     * @param idx - Index of task to delete
+     */
+    public DoneCommand(int idx) {
+        super(null);
+        this.idx = idx;
+    }
+
+    /**
+     * Execute done command on given task and save into tasklist
+     * @param taskList - list containing all existing tasks
+     */
+    @Override
+    public void execute(TaskList taskList) {
+        Task task = taskList.get(idx);
+        taskList.done(this.idx);
+        this.printSuccessfulDoneMessage(task);
+    }
```

Comment:
_Same as before:_

From level-7 task requirements : 'Save the tasks in the hard disk automatically whenever the task list changes.'

Just add your DataStorage.storeTaskList(taskList) command here! (Would need to be in a package to import)
Otherwise a clean implementation 👍

_Suggestion\:_
Since many of the commands need a save to disk operation after executing, 
What I did was use a WritableCommand extends Command abstraction that has 3 methods - execute, run and validate. Execute is implemented to run \: `validate(); run(); storage.saveToDisk(taskList)`.
(ignore the validate() call since your validation logic is inside your parsers)
Then simply extend from WritableCommand for any tasks that need saving after running.